### PR TITLE
changhes for request help form 1

### PR DESCRIPTION
**Pull request by danut3101**
### Requirements for making a pull request

Thank you for contributing to our project! 

Please fill out the template below to help the project maintainers review it as fast as possible and include your contribution to the project.

### What does it fix?

Closes 

Please mention the main changes this PR brings.

### How has it been tested?

Please describe the tests that you ran to verify your changes.

### CODE REVIEW

**Comment by idormenco** [comment]
@danut3101 am testat eu 
las aici ce am gasit (nu trebuie totul fixat in acest PR)
flow solicit ajutor:

**bug**:
click Nevoi speciale (devine checked apare Detaliere nevoi speciale)
click Nevoi speciale (devine un-checked dispare Detaliere nevoi speciale)
completez campurile
apas finalizare

Expected: all good 
Actual: no request is sent (error in console An invalid form control with name='special_request' is not focusable.)

**must have in near future:** 
- [ ]  terms and conditions
- [ ] i18n

**medium:**
- [ ] FE validation nu arata la fel ca BE validation (borders vs popover)
- [ ] FE validation differs from BE validation rules (numar de telefon)

nice to have:
- [ ] spinner la requests

File: app/HelpRequestAccommodationDetail.php [hunk 1]
```diff
@@ -22,7 +22,15 @@
  * @property DateTime|null $created_at
  * @property DateTime|null $updated_at
  * @property ?int $accommodation_id
+ * @property string|null $current_location
+ * @property string|null $known_languages
+ * @property string|null $more_details
+ * @property boolean|null $need_transport
+ * @property boolean|null $dont_need_transport
```
Comment (mihaibrehar) [inline app/HelpRequestAccommodationDetail.php:29]: De ce e nevoie de asta? Nu e suficient daca need_transport este NULL?

File: app/Http/Requests/ServiceRequest.php [hunk 2]
```diff
@@ -28,50 +28,40 @@ public function authorize()
...
+                    $rules['patient-phone'] = ['required', 'max:18', 'min:10', 'regex:/^([0-9\s\-\+\(\)]*)$/'];
+                    $rules['patient-email'] = ['required', 'email', 'string', 'max:255', 'unique:help_requests,patient_email'];
+                    $rules['patient-county'] = ['required', 'exists:ua_regions,id'];
+                    $rules['patient-city'] = ['required', 'exists:ua_cities,id'];
+                    break;
+               case 3:
+                    $rules['requestHelpId'] = ['required', 'numeric', 'gt:0'  ];
+                    $rules['current_location'] = ['required', 'string' ];
+                    $rules['known_languages'] = ['required', 'array','min:1' ];
+                    $rules['more_details'] = [];
```
Comment (mihaibrehar) [inline app/Http/Requests/ServiceRequest.php:45]: Nu se face deloc validare aici?

File: app/Http/Requests/ServiceRequest.php [hunk 3]
```diff
@@ -28,50 +28,40 @@ public function authorize()
...
+                    $rules['special_needs'] = [];
```
Comment (mihaibrehar) [inline app/Http/Requests/ServiceRequest.php:46]: Validare?

File: app/Http/Controllers/RequestServicesController.php [hunk 4]
```diff
@@ -118,14 +176,91 @@ public function submit(ServiceRequest $request)
...
+    /**
+     * @param ServiceRequest $request
+     * @return RedirectResponse
+     */
+    public function submitStep3(ServiceRequest $request)
+    {
+        $request_services_step = $request->get("request_services_step", false);
+        if ($request_services_step == 3) {
+            $requestHelpId = $request->get('requestHelpId', false);
+            if ($requestHelpId > 0) {
```
Comment (mihaibrehar) [inline app/Http/Controllers/RequestServicesController.php:220]: Ar trebui refacut ca sa se reduca nivelurile de imbricare.
```
        if ($request_services_step != 3) {
           return ...;
        }
        
        $requestHelpId = $request->get('requestHelpId', false);
        if (empty($requestHelpId)) {
           return ...
        }
```

File: app/Http/Controllers/RequestServicesController.php [hunk 5]
```diff
@@ -118,14 +176,91 @@ public function submit(ServiceRequest $request)
...
+                            if (!empty($person_in_care_names[$k] ?? "")) {
+                                $help_request_dependant = new HelpRequestDependant();
+                                $help_request_dependant->help_request_id = $requestHelpId;
+                                $help_request_dependant->full_name = ($person_in_care_names[$k] ?? "");
+                                $help_request_dependant->age = ($person_in_care_ages[$k] ?? "");
+                                $help_request_dependant->mentions = ($person_in_care_mentions[$k] ?? "");
+                                $help_request_dependant->save();
+                            }
+                        }
+                    }
```
Comment (mihaibrehar) [inline app/Http/Controllers/RequestServicesController.php:251]: N-ar trebui incluse intr-o tranzactie?

**Comment by gheorghelupu17** [comment]
duplicated by #80

**Pull request closed without merging**